airflow/cli/commands/internal_api_command.py

```diff
@@ -0,0 +1,268 @@
+                f"""\
+                Running the Gunicorn Server with:
+                Workers: {num_workers} {args.workerclass}
+                Host: {args.hostname}:{args.port}
+                Timeout: {worker_timeout}
+                Logfiles: {access_logfile} {error_logfile}
+                Access Logformat: {access_logformat}
+                ================================================================="""
+            )
+        )
+
+        run_args = [
+            sys.executable,
+            "-m",
+            "gunicorn",
+            "--workers",
+            str(num_workers),
+            "--worker-class",
+            str(args.workerclass),
+            "--timeout",
+            str(worker_timeout),
+            "--bind",
+            args.hostname + ":" + str(args.port),
+            "--name",
+            "airflow-internal-api",
+            "--pid",
+            pid_file,
+            "--access-logfile",
+            str(access_logfile),
+            "--error-logfile",
+            str(error_logfile),
+        ]
+
+        if args.access_logformat and args.access_logformat.strip():
+            run_args += ["--access-logformat", str(args.access_logformat)]
+
+        if args.daemon:
+            run_args += ["--daemon"]
+
+        run_args += ["airflow.cli.commands.internal_api_command:cached_app()"]
+
+        # To prevent different workers creating the web app and
+        # all writing to the database at the same time, we use the --preload option.
+        # With the preload option, the app is loaded before the workers are forked, and each worker will
+        # then have a copy of the app
+        run_args += ["--preload"]
+
+        gunicorn_master_proc = None
+
+        def kill_proc(signum, _):
+            log.info("Received signal: %s. Closing gunicorn.", signum)
+            gunicorn_master_proc.terminate()
+            with suppress(TimeoutError):
+                gunicorn_master_proc.wait(timeout=30)
+            if gunicorn_master_proc.poll() is not None:
+                gunicorn_master_proc.kill()
+            sys.exit(0)
+
+        def monitor_gunicorn(gunicorn_master_pid: int):
+            # Register signal handlers
+            signal.signal(signal.SIGINT, kill_proc)
+            signal.signal(signal.SIGTERM, kill_proc)
+
+            # These run forever until SIG{INT, TERM, KILL, ...} signal is sent
+            GunicornMonitor(
+                gunicorn_master_pid=gunicorn_master_pid,
+                num_workers_expected=num_workers,
+                master_timeout=conf.getint("webserver", "web_server_master_timeout"),
+                worker_refresh_interval=conf.getint("webserver", "worker_refresh_interval", fallback=30),
+                worker_refresh_batch_size=conf.getint("webserver", "worker_refresh_batch_size", fallback=1),
+                reload_on_plugin_change=conf.getboolean(
+                    "webserver", "reload_on_plugin_change", fallback=False
+                ),
```

Comment:
Since this a standalone API, should we have separate config for it? In other words, since the API has nothing to do with the webserver, should not we not share configuration between webserver and internal-api?

Reply:
Fair point - removed the config options and used some defaults - we may introduce them later, when we are closer to making a "release" of Internal API.